common/mixins.py

```diff
@@ -32,7 +32,8 @@ def clean(self):
             slugify_field_value = getattr(self, self.slugify_field)
             self.slug = slugify(slugify_field_value)
 
-            if self.__class__.objects.filter(slug=self.slug).exists():
+            # If pk exists, object exists in db and is being edited by user.
+            if self.__class__.objects.filter(slug=self.slug).exists() and not self.pk:
```

Comment:
@andela-kanyanwu: For update, there's possibility of changing the name to something that would resolve to `slug` field value that exists already. For that case, the error message displayed would be in reference to the `slug` field which I don't think is the desired one; is it?